crates/red_knot_python_semantic/src/types/builder.rs

```diff
@@ -360,7 +362,7 @@ mod tests {
     #[test]
     fn build_union_bool() {
         let db = setup_db();
-        let bool_ty = builtins_symbol_ty(&db, "bool");
+        let bool_ty = KnownClass::Bool.to_class(&db);
```

Comment:
Ah. @carljm, I think this might actually reveal a bug in `main`. The union of `Literal[True] | Literal[False]` should surely simplify to `Instance("bool")` rather than `Class("bool")`. If an object has type `Literal[True]`, it indicates that it the object is an instance of `bool`, not that it is the `bool` class itself.

The fact that the new idiom in this PR makes this bug more obvious is definitely a point in favour of us landing it @Slyces!!

Reply:
Great catch, definitely didn't see it myself!

Reply:
(It doesn't need to be fixed in this PR, since I guess it'll require other changes elsewhere that your PR currently doesn't touch! Though it might actually be a pretty small change — if so, it's probably okay to fix it here if you feel like it 😆)

Reply:
Might prefer a separate PR if that's ok!

Reply:
https://github.com/astral-sh/ruff/issues/13642